### PR TITLE
test: add tests for rpcProvider & update docs

### PR DESCRIPTION
**Pull request by notV4l**
- rel : #589 
- add tests for rpcProvider
- update getEvents docs

## Checklist:

- [x] Performed a self-review of the code
- [ ] Rebased to the last commit of the target branch (or merged it into my branch)
- [X] Linked the issues which this PR resolves
- [X] Documented the changes
- [x] Updated the docs (www)
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

**Comment by PhilippeR26** [comment]
As the API doc is now automatically generated, take care that no more files shall be added/modified in  www/doc/API.

**Comment by ivpavici** [comment]
Up

> As the API doc is now automatically generated, take care that no more files shall be added/modified in www/doc/API.

updated the PR template to remove this section!

**Review by tabaktoni — APPROVED** [review]
LGTM (except the commented part with docs)

**Review by ivpavici — APPROVED** [review]

File: www/docs/API/Provider/rpcProvider.md [hunk 1]
```diff
@@ -299,12 +299,12 @@ Gets all the events filtered
 
 ```typescript
 type EventFilter = {
-  fromBlock: string;
-  toBlock: string;
+  from_block: string;
```
[the remark this answers is not in the document]
Comment (notV4l) [reply www/docs/API/Provider/rpcProvider.md]: done

**Pull request merged**